Assets/Scripts/CaveV2/CaveGenComponentV2.cs

```diff
@@ -767,6 +772,10 @@ private CaveGraphV2 GenerateCaveGraph(CaveGenParameters caveGenParams, Bounds bo
                             edge.IsBlocked = !nodeBIsTowardsStart;
                         }
                     }
+                    
+                    if(allEdgesNotTowardStartBlocked) {
+                        nodeA.NodeType = CaveNodeType.Challenge;
```

Comment:
This seems like it could result in an unpredictable frequency of challenge rooms. Have you been seeing a consistent amount in each level?

Reply:
It is stable, but not the same everytime, 1-3 every level most of the time 2. If we wanted to change that, we need to think of where we want the challenge rooms to spawn then, maybe they are just normal rooms with guaranteed spawns and chests